### ISSUE TITLE
Support for caching changes (in localstorage?)

### ISSUE THREAD

**Issue by jamespohalloran** [tinacms/tina-graphql-gateway#218]
## Summary

With tinacms, you can use the `useLocalStorageCache` hook to persist changes between page loads in local storage.

E.g: https://github.com/tinacms/tinacms.org/blob/master/components/layout/InlineGithubForm.tsx#L18

To my knowledge, there's not a defined way to cache changes locally with Tina Cloud / Graphql Gateway

### Basic example

The `useLocalStorageCache` hook takes in a form. This won't work well with the TinaCloud Client, where the form is extended in a `formify` callback

```
  const [payload, isLoading] = useGraphqlForms({
    query,
    variables: {
      relativePath: `index.md`,
    },
    formify: ({ createForm, formConfig }) => {
     // ...
      return createForm(formConfig);
    },
  });
```

We can't use hooks within a callback, so this won't work as is.

### Motivation

I'm not sure if using the existing `useLocalStorageCache` is the answer, but supporting local caching will be a big editing improvement.

**Comment by jeffsee55** [comment]
Yeah, we tend to have a lot of super convenient hooks whose underlying logic isn't easy to use. Though this isn't a hook from Tina. Do you think it's too cumbersome to just do this from `form.subscribe`?

**Comment by jamespohalloran** [comment]
I think ideally for me, this caching behavior is something the Client could just do automatically under the hood. 
I'd have to see an example using form.subscribe stuff, but it's probably not too big of a deal to call manually as a first step.

**Comment by Enigmatical** [comment]
* Experiment with adding events for Form creation and destruction (`unstable:` or `experimental:` namespace)
* Wrap the localStorage or browserStorage behavior in a plugin that can add the watcher to forms through those events
* Register to `starter` as a POC
